test/Microsoft.AspNetCore.Mvc.FunctionalTests/RazorViewLocationSpecificationTest.cs

```diff
@@ -93,5 +94,20 @@ public RazorViewLocationSpecificationTest(MvcTestFixture<RazorWebSite.Startup> f
             // Assert
             Assert.Equal(expected, body.Trim(), ignoreLineEndingDifferences: true);
         }
+
+        [Fact]
+        public async Task PartialLayout_ThrowsIfRequiredSectionMissing()
+        {
+            // Arrange
+            var path = "http://localhost/PartialViewEngine/ViewPartialMissingSection";
+
+            //Act
+            var ex = await Assert.ThrowsAsync<InvalidOperationException>(() => Client.GetAsync(path));
```

Comment:
This isn't a good functional test: Should never rely on `Exception`s across the client / server boundary. Could instead catch the expected `InvalidOperationException` in an `ExceptionFilterAttribute` and return the expected content. Bit uglier and definitely not recommended for production use (because the view could in general be left in a weird state) but could also use `try` / `catch` in the layout file.